backend/database/supabase_middleman.py

```diff
@@ -244,3 +246,35 @@ def update_entry():
     """
     Updates an entry in a table
     """
+
+
+def authenticate(jwt: str) -> str:
+    """Authenticates a user by supabase jwt (Authorization header)
+
+    Args:
+        jwt (str): Supabase jwt
+
+    Raises:
+        HTTPException: If the token passed is invalid
+
+    Returns:
+        str: User ID
+    """
+    unauth_error = HTTPException(status_code=401, detail="Invalid token")
+
+    # separate case to prevent any kind of session leakage because of
+    # behaivior of get_user when there is no token passed.
+    if not jwt:
+        raise unauth_error
+
+    try:
+        data = supabase.auth.get_user(jwt)
+    except Exception as e:
+        logging.warning("Error when authenticating user", exc_info=e)
+
+        raise unauth_error from e
+
+    if data is None:
+        raise unauth_error
+
+    return data.user.id
```

Comment:
NIT: could be written nicer

```suggestion
    return unauth_error if data is None else return data.user.id
```

although you could argue that this is more readable

Reply:
`unauth_error` is raised, not returned

Reply:
sorry quickly read it, let me fix it

Reply:
fixed

Reply:
(idk if its valid syntax tho, again pretty nit picky)